### PR TITLE
wip: Build: Browserify => parcel (refs #9478)

### PR DESCRIPTION
**Pull request by Aladdin-ADD**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Other, please explain:

**What changes did you make? (Give an overview)**
replaced Browserify with parcel

**Is there anything you'd like reviewers to focus on?**

+ set babel browsers  targets to `defaults`. Is there some best pratice about it?
+ removed babel-polyfill, but add [polyfill.io](https://polyfill.io) in https://github.com/eslint/eslint.github.io/pull/477, that would reduce the size for modern browsers. 😄

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "presets": [
+    [
+      "env",
+      {
+        "targets": {
+          "browsers": [
+            ">1%",
+            "last 2 versions"
```
[the remark this answers is not in the document]
Comment (not-an-aardvark) [reply .babelrc]: I think we intentionally support IE11 in the ESLint demo on `eslint.org` at the moment at the moment.

**Comment by platinumazure** [comment]
Probably a good idea to create an issue for further discussion, since this is basically a core change.

Personally, I'd like to know what problems exist with the current setup and how parcel (specifically) solves them. But let's discuss in an issue :smile:

**Comment by Aladdin-ADD** [comment]
l had created an issue #9478 , use rollup/webpack.  `parcel`, the emerging bundler, can be another alternative to consider in this situation. @platinumazure

File: .babelrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aladdin-ADD) [reply .babelrc]: well, I think the `defaults` includes IE11 already. (http://browserl.ist/)

[the remark this answers is not in the document]
Comment (g-plane) [reply .babelrc]: The `defaults` includes some `0%` browsers.

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "presets": [
+    [
+      "env",
+      {
+        "targets": {
+          "browsers": [
+            "defaults"
```
Comment (kaicataldo) [inline .babelrc]: What does this do? I'm not seeing anything about it in the docs.

File: package.json [hunk 3]
```diff
@@ -76,6 +76,7 @@
   "devDependencies": {
     "babel-core": "^6.26.3",
     "babel-polyfill": "^6.26.0",
+    "babel-preset-env": "^1.7.0",
     "babel-preset-es2015": "^6.24.1",
```
Comment (kaicataldo) [inline package.json]: This is no longer needed if we're using preset-env.

File: package.json [hunk 4]
```diff
@@ -76,6 +76,7 @@
...
     "babelify": "^8.0.0",
```
Comment (kaicataldo) [inline package.json]: We should also remove this and `browserify`

File: package.json [hunk 5]
```diff
@@ -76,6 +76,7 @@
   "devDependencies": {
     "babel-core": "^6.26.3",
     "babel-polyfill": "^6.26.0",
```
Comment (kaicataldo) [inline package.json]: This should no longer needed if we're using preset-env.

**Pull request closed without merging**